### PR TITLE
make LR parameters customizable in training workflow

### PR DESCRIPTION
**Pull request by mayasheth**
tested with no regularization, L1, L2, and elastic net settings

@atancoder

### CODE REVIEW

File: config/config_models_test.tsv [hunk 1]
```diff
@@ -1,3 +1,13 @@
-model	dataset	ABC_directory	feature_table	polynomial
-DNase_megamap_8features	DNase_megamap		resources/feature_tables/final_feature_set_DNase_hic.tsv	False 
-DNase_intact_8features	DNase_intact		resources/feature_tables/final_feature_set_DNase_hic.tsv	False
+model	dataset	ABC_directory	feature_table	polynomial	override_params
+DNase_megamap_noReg	DNase_megamap		resources/feature_tables/all_features_DNase_hic.tsv	False 	
+DNase_megamap_L1	DNase_megamap		resources/feature_tables/all_features_DNase_hic.tsv	False 	{'solver': 'saga', 'penalty': 'l1'}
```
Comment (atancoder) [inline config/config_models_test.tsv:3]: I realize this is important for us to recognize what are the best hyperparameters for our logistic regression model. Do we expect users to want to choose their own hyperparameters? Or are we going to find the best ones and make them the default?

File: workflow/rules/train_model.smk [hunk 2]
```diff
@@ -105,6 +125,40 @@ rule train_model:
 			--feature_table_file {input.feature_table} \
 			--out_dir {params.out_dir} \
 			--polynomial {params.polynomial} \
-			--epsilon {params.epsilon}
+			--epsilon {params.epsilon} \
+			--params_file {input.model_params}
 		"""
 		
+# compare cv-performance on training data across all models (note, this is not the true benchmarking performance CRISPR elements not overlapping prediction elements aren't considered)  
+rule gather_model_performances:
```
Comment (atancoder) [inline workflow/rules/train_model.smk:133]: For the future, you can consider splitting the actual model training components from the best model selection components into separate snakemake modules to make things cleaner.

File: workflow/scripts/compare_all_models.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                                 'precision', 'precision_95CI_low', 'precision_95CI_high' ])
+    
+    # iterate through rows of model config and add results to final df
+    for row in model_config.itertuples(index=False):
+         res_row = performance_summary(row.model, row.dataset, model_name, out_dir)
+         df = pd.concat([df, res_row])
+
+    # sort table by AUPRC
+    df = df.sort_values(by='AUPRC', ascending=False)
+    df.to_csv(output_file, sep = '\t', index=False)
```
Comment (atancoder) [inline workflow/scripts/compare_all_models.py:49]: it would be good to share a snippet of the output from your changes in the PR summary/test plan

**Review by atancoder — APPROVED** [review]

File: config/config_models_test.tsv (hunk 1 above)
Comment (mayasheth) [reply config/config_models_test.tsv:3]: No idea who is going to use this besides us... the default parameters are already specified in config_training which is more transparent and explicit than having it hardcoded. The override_params can just be used if the user wants to change one of those for any particular model. The whole point of this workflow is to compare different model architectures so I feel like it;s a valid things to include!

**Pull request merged**